packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -968,11 +990,18 @@ class _RepeatingSimulation extends Simulation {
   final double min;
   final double max;
   final bool reverse;
+  final int? repeatCount;
   final _DirectionSetter directionSetter;
 
   final double _periodInSeconds;
   final double _initialT;
 
+  /// If [reverse] is true, the duration of a full cycle is doubled.
+  late final int _effectiveRepeatCount = repeatCount! * (reverse ? 2 : 1);
+
+  late final double _exitTimeInSeconds =
+      (_effectiveRepeatCount * _periodInSeconds) - _initialT;
```

Comment:
```suggestion
  late final double _exitTimeInSeconds = (_effectiveRepeatCount * _periodInSeconds) - _initialT;
```

Small nit: I think this should be one line. There's a similar example from the [style guide](https://github.com/flutter/flutter/blob/38af6be9fb29bb87b9b2fb7853fe22b4287ea917/docs/contributing/Style-guide-for-Flutter-repo.md#prefer-a-maximum-line-length-of-80-characters):

```dart
// BAD (breaks after assignment operator)
final List<FooBarBaz> _members =
  <FooBarBaz>[const Quux(), const Qaax(), const Qeex()];

// BETTER (only slightly goes over 80 chars)
final List<FooBarBaz> _members = <FooBarBaz>[const Quux(), const Qaax(), const Qeex()];
```

Reply:
hmm thanks for pointing out the style guidelines. I have updated as per suggestion.